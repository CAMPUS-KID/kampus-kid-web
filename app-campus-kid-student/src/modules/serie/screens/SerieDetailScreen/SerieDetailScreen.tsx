import { useFetchSerieDetail } from '@modules/serie/hooks/useFetchSerieDetail';
import { useBuildStyles } from '@shared_hooks';
import { buildImageURL } from '@shared_utils';
import React from 'react';
import { ActivityIndicator, Image, Text, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { styleSheet } from './styles';

export interface Props {
  route: {
    params: {
      serieId: number;
    };
  };
}

const SerieDetailScreen = ({
  route: {
    params: { serieId }
  }
}: Props) => {
  const styles = useBuildStyles(styleSheet);
  const { isFetchingSerieDetail, fetchSerieDetailError, serieDetail } =
    useFetchSerieDetail(serieId);

  if (isFetchingSerieDetail) {
    return (
      <SafeAreaView>
        <ActivityIndicator animating size="large" />
      </SafeAreaView>
    );
  }
  if (fetchSerieDetailError) {
    return (
      <SafeAreaView>
        <Text style={styles.errorText}>{fetchSerieDetailError.message}</Text>
      </SafeAreaView>
    );
  }
  return (
    <SafeAreaView>
      <View>
        <Image
          style={styles.headerImage}
          source={{ uri: buildImageURL(serieDetail.poster_path) }}
        />
        <Text style={styles.titleText}>{serieDetail.name}</Text>
      </View>
    </SafeAreaView>
  );
};

export default SerieDetailScreen;
