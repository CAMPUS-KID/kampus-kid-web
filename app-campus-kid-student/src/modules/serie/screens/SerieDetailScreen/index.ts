export { default as SerieDetailScreen } from './SerieDetailScreen';
