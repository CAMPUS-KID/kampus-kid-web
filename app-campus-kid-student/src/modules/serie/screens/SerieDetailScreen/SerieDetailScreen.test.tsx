import { wait } from '@test_shared';
import React from 'react';
import { render } from 'react-native-testing-library';
import { act } from 'react-test-renderer';
import SerieDetailScreen, { Props } from './SerieDetailScreen';

describe('SerieDetailScreen', () => {
  let props: Props;
  beforeEach(() => {
    props = {
      route: { params: { serieId: 100 } }
    };
  });
  it('should render', async () => {
    const { toJSON } = render(<SerieDetailScreen {...props} />);
    await act(() => wait());
    expect(toJSON()).toMatchSnapshot();
  });
});
