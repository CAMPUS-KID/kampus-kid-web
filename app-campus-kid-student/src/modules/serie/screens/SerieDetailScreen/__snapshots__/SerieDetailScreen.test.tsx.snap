// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SerieDetailScreen should render 1`] = `
<RNCSafeAreaView>
  <View>
    <Image
      source={
        Object {
          "uri": "undefined/pTEFqAjLd5YTsMD6NSUxV6Dq7A6.jpg",
        }
      }
      style={
        Object {
          "height": 300,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#181818",
          "fontFamily": "Roobert-Bold",
          "fontSize": 28,
          "textAlign": "center",
        }
      }
    >
      All of Us Are Dead
    </Text>
  </View>
</RNCSafeAreaView>
`;
